byway/src/main/webapp/destinations.js

```diff
@@ -189,8 +192,7 @@ function updateStartDestination(locationData){
 * fetches data from servlet
 */
 function fetchDestinations(){
-  let promise= fetch('/api/destinations').then(response => response.json());
-  return promise;
+  return fetch('/api/destinations?tripKey=' + tripKey).then(response => response.json());
```

Comment:
When building query parameters prefer to use `URLSearchParams`, as this will handle proper escaping and makes it easier if you start needing to set multiple parameters.

Reply:
fixed